jumpscale/sals/nettools/__init__.py

```diff
@@ -32,117 +47,167 @@ def tcp_connection_test(ipaddr: str, port: int, timeout: Optional[int]):
     """
     conn = None
     try:
-        conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        if timeout:
-            conn.settimeout(timeout)
-        try:
-            conn.connect((ipaddr, port))
-        except BaseException:
-            return False
-    finally:
-        if conn:
-            conn.close()
-    return True
+        conn = socket.create_connection((ipaddr, port), timeout)
+    except (socket.gaierror, socket.herror):
+        # raised for address-related errors
+        raise
+    except OSError:
+        # (ConnectionRefusedError, socket.timeout, OSError)
+        return False
+    else:
+        conn.close()
+        return True
 
 
-def udp_connection_test(ipaddr: str, port: int, timeout=1, message=b"PING"):
-    """tests udp connection on specified port by sending specified message
+def udp_connection_test(ipaddr: str, port: int, timeout: Optional[int] = 1, message: Optional[bytes] = b"") -> bool:
+    """tests udp connection on specified port by sending specified message and expecting
+    to receive at least one byte from the socket as an indicator of connection success
 
     Args:
         ipaddr (str): ip address
         port (int): port number
         timeout (int, optional): time before the connection test fails. Defaults to None.
         message (str, optional): message to send. Defaults to b"PING"
 
+    Raises:
+        ValueError: raises if invalid ip address was used
+
     Returns:
         bool: True if the test succeeds, False otherwise
     """
-    conn = socket.socket(type=socket.SOCK_DGRAM)
+    try:
+        ip = ipaddress.ip_address(ipaddr)
+    except ValueError:
+        raise
```

Comment:
we can log the exception and return

Reply:
did you mean to log the exception and raise it?
because I think returning, in case of an invalid IP address passed in, is not a legitimate option, for the use cases of this function.

Reply:
just like that one https://github.com/threefoldtech/js-ng/pull/531#discussion_r559578111